gammapy/irf/psf_table.py

```diff
@@ -753,3 +758,412 @@ def _get_1d_table_psf(self, energy_index, **kwargs):
+            Offset in the field of view. Default theta = 0 deg
+        interp_kwargs : dict
+            Option for interpolation for `~scipy.interpolate.RegularGridInterpolator`
+            
+        Returns
+        -------
+        table : `~astropy.table.Table`
+            Table with two columns: offset, value
+        """
+        
+        # Defaults
+        theta = theta or Angle(0, 'deg')
+        
+        psf_value = self.evaluate(energy, theta, interp_kwargs=interp_kwargs).squeeze()
+        for v in psf_value.value:
+            if v != v or v == 0:
+                return None
+        table_psf = TablePSF(self.rad_center(), psf_value, **kwargs)
+        
+        return table_psf
+
+    def containment_radius(self, energy, theta=None, fraction=0.68, interp_kwargs=None):
+        """Containment radius.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
+            Energy
+        theta : `~astropy.coordinates.Angle`
+            Offset in the field of view. Default theta = 0 deg
+        fraction : float
+            Containment fraction. Default fraction = 0.68
+
+        Returns
+        -------
+        radius : `~astropy.units.Quantity`
+            Containment radius in deg
+        """
+
+        # Defaults
+        theta = theta or Angle(0, 'deg')
+        if energy.ndim == 0:
+            energy = Quantity([energy.value], energy.unit)
+        if theta.ndim == 0:
+            theta = Quantity([theta.value], theta.unit)
+        
+        unit = None
+        radius = np.zeros((energy.size, theta.size))
+        for e in range(energy.size):
+            for t in range(theta.size):
+                psf = self.to_table_psf(energy[e], theta[t], interp_kwargs)
+                if psf == None:
+                    radius[e, t] = np.nan
+                    continue
+                r = psf.containment_radius(fraction)
+                radius[e, t] = r.value
+                unit = r.unit
+        return Quantity(radius.squeeze(), unit)
+        
+    def plot_containment_vs_energy(self, fractions=[0.68, 0.95],
+                                   thetas=Angle([0, 1], 'deg'), ax=None, **kwargs):
+        """Plot containment fraction as a function of energy.
+        """
+        import matplotlib.pyplot as plt
+
+        ax = plt.gca() if ax is None else ax
+
+        energy = Energy.equal_log_spacing(
+            self.energy_lo[0], self.energy_hi[-1], 100)
+
+        for theta in thetas:
+            for fraction in fractions:
+                radius = self.containment_radius(energy, theta, fraction).squeeze()
+                label = '{} deg, {:.1f}%'.format(theta, 100 * fraction)
+                ax.plot(energy.value, radius.value, label=label)
+
+        ax.semilogx()
+        ax.legend(loc='best')
+        ax.set_xlabel('Energy (TeV)')
+        ax.set_ylabel('Containment radius (deg)')
+
+    def plot_psf_vs_rad(self, filename=None, theta=Angle(0, 'deg'), energy=Quantity(1, 'TeV')):
```

Comment:
Is it useful to have `plot_psf_vs_rad` on the `PSF3D` class?
Can't you use or improve the [EnergyDependentTablePSF.plot_psf_vs_theta](https://gammapy.readthedocs.io/en/latest/api/gammapy.irf.EnergyDependentTablePSF.html#gammapy.irf.EnergyDependentTablePSF.plot_psf_vs_theta) method to make these plots?